File: src/main/java/ru/stnk/RestTestAPI/service/SheduleDeleteRowsInCoinTables.java
package ru.stnk.RestTestAPI.service;


import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Component;
import ru.stnk.RestTestAPI.entity.coins.QuoteBinFiveMinute;
import ru.stnk.RestTestAPI.entity.coins.QuoteBinOneMinute;
import ru.stnk.RestTestAPI.repository.QuoteBinFiveMinuteRepository;
import ru.stnk.RestTestAPI.repository.QuoteBinOneMinuteRepository;

import java.util.List;

@Component
public class SheduleDeleteRowsInCoinTables {
    @Autowired
    private QuoteBinOneMinuteRepository quoteBinOneMinuteRepository;

    @Autowired
    private QuoteBinFiveMinuteRepository quoteBinFiveMinuteRepository;

    @Scheduled(fixedDelay = 60000)
    public void deleteRows () {
        List<QuoteBinOneMinute> oneMinuteList = quoteBinOneMinuteRepository.findAll();
        List<QuoteBinFiveMinute> fiveMinuteList = quoteBinFiveMinuteRepository.findAll();

        if (oneMinuteList.size() > 500) {
            for (int i = 0; i < oneMinuteList.size() - 499; i++) {
                quoteBinOneMinuteRepository.delete(oneMinuteList.get(i));
            }
        }

        if (fiveMinuteList.size() > 500) {
            for (int i = 0; i < fiveMinuteList.size() - 499; i++) {
                quoteBinFiveMinuteRepository.delete(fiveMinuteList.get(i));
            }
        }
    }
}

File: src/main/kotlin/ru/stnk/resttestapi/service/SheduleDeleteRowsInCoinTables.kt
package ru.stnk.resttestapi.service


import org.springframework.beans.factory.annotation.Autowired
import org.springframework.scheduling.annotation.Scheduled
import org.springframework.stereotype.Component
import ru.stnk.resttestapi.repository.QuoteBinFiveMinuteRepository
import ru.stnk.resttestapi.repository.QuoteBinOneMinuteRepository

@Component
class SheduleDeleteRowsInCoinTables {
    @Autowired
    private val quoteBinOneMinuteRepository: QuoteBinOneMinuteRepository? = null

    @Autowired
    private val quoteBinFiveMinuteRepository: QuoteBinFiveMinuteRepository? = null

    @Scheduled(fixedDelay = 60000)
    fun deleteRows() {
        val oneMinuteList = quoteBinOneMinuteRepository!!.findAll()
        val fiveMinuteList = quoteBinFiveMinuteRepository!!.findAll()

        if (oneMinuteList.size > 500) {
            for (i in 0 until oneMinuteList.size - 499) {
                quoteBinOneMinuteRepository.delete(oneMinuteList[i])
            }
        }

        if (fiveMinuteList.size > 500) {
            for (i in 0 until fiveMinuteList.size - 499) {
                quoteBinFiveMinuteRepository.delete(fiveMinuteList[i])
            }
        }
    }
}
